app/jobs/consume_assessments_job.rb

```diff
@@ -14,18 +14,37 @@ def perform
       begin
         message = JSON.parse(msg)&.slice('threshold_condition_hash', 'reported_symptoms_array',
                                          'patient_submission_token', 'experiencing_symptoms', 'response_status')
-
         # Invalid message
         if message.nil?
           Rails.logger.info 'ConsumeAssessmentsJob: skipping nil message...'
           queue.commit
           next
         end
 
-        patient = Patient.where(purged: false).find_by(submission_token: message['patient_submission_token'])
+        unless message['response_status'].in? %w[opt_out opt_out]
+          patient = Patient.where(purged: false).find_by(submission_token: message['patient_submission_token'])
+        end
 
-        # Perform patient lookup for old submission tokens
-        if patient.nil?
+        if message['response_status'].in? %w[opt_out opt_in]
+          # When an opt_in or opt_out response_status is posted to us the patient_submission_token value is popuated with
+          # a flow execution id, this is because a monitoree may send STOP/START outside the context of an assessment and
+          # therefore the patient.submission_token will not be available. We get the responder associated with the opt_in
+          # or opt_out phone number by requesting the phone number who sent the message in the associated flow execution id
+          phone_numbers = TwilioSender.get_phone_numbers_from_flow_execution(message['patient_submission_token'])
+          if phone_numbers.nil?
+            Rails.logger.info 'ConsumeAssessmentsJob: skipping nil flow execution'
+            queue.commit
+            next
+          end
+          monitoree_number = phone_numbers[:monitoree_number]
+          sara_number = phone_numbers[:sara_number]
+          # Handle BlockedNumber manipulation here in case no monitorees are associated with this number
+          BlockedNumber.create(phone_number: monitoree_number) if message['response_status'] == 'opt_out'
```

Comment:
For consistency - should we store these numbers in the same format we store the `primary_telephone`?

Reply:
Twilio API does return in the same format -- but it's a good point, we can put it through the PhoneLib.parse function just in case Twilio ever changes the way that they send phone numbers over their API

Reply:
Yeah I think that's a good idea - something to guarantee it's always consistent in case someone stands up Sara Alert with a different provider as well.